code/defines/obj.dm

```diff
@@ -198,6 +198,147 @@
+			J = flags_to_job(player.client.prefs.job_medsci_high,MEDSCI)
+		else if(player.client.prefs.job_civilian_high)
+			J = flags_to_job(player.client.prefs.job_civilian_high,CIVILIAN)
+		else
+			continue //They don't have a high pref!
+
+		if(!J)
+			continue //sanity
+		crystal_ball[J] += 1
+
+/datum/controller/occupations/proc/flags_to_job(var/flags, var/department)
+	//Takes a flag and returns a job. First it tries to get it from a hardcoded list for speed, but if that doesn't work we just loop through (for forward compatibility)
+	switch(department)
+		if(CIVILIAN)
+			switch(flags)
+				if(1)
+					return "Head of Personnel" //datum/job/hop
+				if(2)
+					return "Bartender" //datum/job/bartender
+				if(4)
+					return "Botanist" //datum/job/hydro
+				if(8)
+					return "Chef" //datum/job/chef
+				if(16)
+					return "Janitor" //datum/job/janitor
+				if(32)
+					return "Librarian" //datum/job/librarian
+				if(64)
+					return "Quartermaster" //datum/job/qm
+				if(128)
+					return "Cargo Technician" //datum/job/cargo_tech
+				if(256)
+					return "Shaft Miner" //datum/job/mining
+				if(512)
+					return "Internal Affairs" //datum/job/lawyer
+				if(1024)
+					return "Chaplain" //datum/job/chaplain
+				if(2048)
+					return "Clown" //datum/job/clown
+				if(4096)
+					return "Mime" //datum/job/mime
+				if(8192)
+					return "Assistant" //datum/job/assistant
+				if(16384)
+					return "Trader" //datum/job/trader
+		if(MEDSCI)
+			switch(flags)
+				if(1)
+					return "Research Director" //datum/job/rd
+				if(2)
+					return "Scientist" //datum/job/scientist
+				if(4)
+					return "Chemist" //datum/job/chemist
+				if(8)
+					return "Chief Medical Officer" //datum/job/cmo
+				if(16)
+					return "Medical Doctor" //datum/job/doctor
+				if(32)
+					return "Geneticist" //datum/job/geneticist
+				if(64)
+					return "Virologist" //datum/job/virologist
+				if(128)
+					return "Paramedic" //datum/job/paramedic
+		if(ENGSEC)
+			switch(flags)
+				if(1)
+					return "Captain" //datum/job/captain
+				if(2)
+					return "Head of Security" //datum/job/hos
+				if(4)
+					return "Warden" //datum/job/warden
+				if(8)
+					return "Detective" //datum/job/detective
+				if(16)
+					return "Security Officer" //datum/job/officer
+				if(32)
+					return "Chief Engineer" //datum/job/chief_engineer
+				if(64)
+					return "Station Engineer" //datum/job/engineer
+				if(128)
+					return "Atmospherics Tech." //datum/job/atmos
+				if(256)
+					return "Roboticist" //datum/job/roboticist //Why is this here? Mystery
+				if(512)
+					return "A.I." //datum/job/ai
+				if(1024)
+					return "Cyborg" //datum/job/cyborg
+				if(2048)
+					return "Mobile MMI" //datum/job/mommi
+				if(4096)
+					return "Mechanic" //datum/job/mechanic
+	//Still haven't found anything? Let's try this the hard way.
+	var/list/searchable_jobs = typesof(/datum/job) - /datum/job
+	for(var/datum/job/J in searchable_jobs)
```

Comment:
You're looking for instances of `/datum/job` but you're iterating over a list of type paths, so this loop never runs. I suppose this is what prompted you to make the huge switch/case above in the first place?

Reply:
No, I made the above switch because I thought there was a risk in 60-pop of people repeatedly hitting "predict manifest" to "refresh". I was worried looping through 33 jobs times 60 people multiple times a second might be resource intensive for the lobby. Since testing never had any invalid job types I never tested the "futureproof" code at all.

I guess I'll have to instantiate each path down there as they get looped through... all the more reason to use the switch, right?

Reply:
Talking performance without numbers is rarely wise. A smarter approach would be to cache the window's HTML after the first time it's accessed, mark it as dirty on the next preference update, re-generate and cache it next time the window is accessed, and so on. I'm pretty sure the current crew manifest already does this.
In case it's not clear, I __really__ don't like the huge switch.

Reply:
If you insist, then. It shouldn't be that hard to modify, although I'm a little suspicious of loops within loops...

I'd be more insistent if it was something happening during regular gameplay, but maybe it won't be so bad since nothing else is being processed during the lobby phase.

Reply:
>I guess I'll have to instantiate each path down there as they get looped through.

Provided you're not looking at lists, `initial` should work.